### PR TITLE
push Appyter

### PR DESCRIPTION
**Pull request by jchpan**
Try to push Juncheng' Appyter to main catalog

### CODE REVIEW

**Comment by jchpan** [comment]
fix issues

File: appyters/Dimensionality_Reduction/appyter.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "version": "0.0.1",
+  "description": "An appyter for to perform visualization using PCA, t-SNE, and UMAP methods.",
+  "image": "scatter_screenshot.png",
+  "authors": [
+    {
+      "name": "Juncheng Pan",
+      "email": "pan_juncheng@yahoo.com"
+    }
+  ],
+  "url": "https://github.com/MaayanLab/Harmonizome-Data-Processing-Appyters",
```
[the remark this answers is not in the document]
Comment (jchpan) [reply appyters/Dimensionality_Reduction/appyter.json]: fixed it

File: appyters/Dimensionality_Reduction/appyter.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "https://raw.githubusercontent.com/MaayanLab/appyter-catalog/main/schema/appyter-validator.json",
+  "name": "Dimension_Reduction_Appyter",
```
[the remark this answers is not in the document]
Comment (jchpan) [reply appyters/Dimensionality_Reduction/appyter.json]: fixed it

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%appyter code_exec\n",
+    "rnaseq_data_filename = \"{{rnaseq_data_filename.value}}\"\n",
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:136]: Please use `{{rnaseq_data_filename}}`, quotes will be added automatically in a secure manner, same thing with the other fields.

**Review by u8sand — CHANGES_REQUESTED** [review]

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb [hunk 4]
```diff
@@ -202,43 +203,18 @@
...
-    "# Match samples between the metadata and the datasets\n",
-    "try:\n",
-    "    check_df(meta_df, meta_class_column_name)\n",
-    "except:\n",
-    "    print(f\"Error! Column '{meta_class_column_name}' is not in metadata\")"
+    "    return temp_df\n",
+    "\n",
+    "expr_df = load_file(rnaseq_data_filename, index_col=0)\n",
+    "meta_df = load_file(meta_data_filename)"
    ]
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:218]: Don't forget to add
```python
assert meta_class_column_name in meta_df.columns, f"Error! Column '{meta_class_column_name}' is not in metadata"
```

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb [hunk 5]
```diff
@@ -332,7 +308,7 @@
     "\n",
     "def model_t_SNE_feature():\n",
     "    numbers = 3 if plot_3D == \"3D\" else 2\n",
-    "    tsne = TSNE(init='pca', n_components=numbers, learning_rate='auto')\n",
+    "    tsne = TSNE(n_jobs = 4, n_components=numbers)\n",
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:311]: A good way to make n_jobs system agnostic is

```python
import multiprocessing as mp

#...
n_jobs=mp.cpu_count()
```

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb [hunk 6]
```diff
@@ -23,7 +23,8 @@
     "import matplotlib.pyplot as plt\n",
     "%matplotlib inline\n",
     "from sklearn.decomposition import PCA\n",
-    "from sklearn.manifold import TSNE\n",
+    "# from sklearn.manifold import TSNE\n",
+    "from MulticoreTSNE import MulticoreTSNE as TSNE\n",
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:27]: Of course this package needs to be added to `requirements.txt`

File: appyters/dimensionality_reduction_visualization/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+pandas
+matplotlib
+sklearn
+seaborn
+IPython
+bokeh
+umap-learn
+cmake==3.18.4
+MulticoreTSNE
+react-scatter-board @ git+https://github.com/maayanlab/react-scatter-board@v2
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/requirements.txt:15]: Are you using v2 or v2.1?

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb [hunk 8]
```diff
@@ -0,0 +1,512 @@
...
+   "outputs": [],
+   "source": [
+    "%%appyter code_exec\n",
+    "rnaseq_data_filename = {{rnaseq_data_filename}}\n",
+    "meta_data_filename = {{meta_data_filename}}\n",
+    "meta_class_column_name = {{meta_class_column_name}}\n",
+    "\n",
+    "plot_3D = \"{{plot_3D.value}}\"\n",
+    "clustering_topk = 50\n",
+    "method = \"{{visualization_method.value}}\"\n"
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:143]:
```python
plot_3D = "{{plot_3D}}"
method = "{{visualization_method}}"
```

Should work here, you should only need .value for jinja2 comparisons

Comment (jchpan) [reply appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb]: removed it

**Review by u8sand — APPROVED** [review]

**Pull request merged**